spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/request/Request.java

```diff
@@ -0,0 +1,331 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.client.request;
+
+import org.candlepin.ApiException;
+import org.candlepin.Pair;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import okhttp3.Call;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+
+
+/**
+ * The Request class provides a fluent-style builder API for creating generic requests. The Request
+ * instance needs an ApiClient instance to perform the actual work, and can be provided either
+ * directly during construction, or can be pulled automatically from a client-level ApiClient
+ * object.
+ */
+public class Request {
+
+    private static final Pattern PATH_PARAM_REGEX = Pattern.compile("\\{([^}]+)\\}");
+
+    private final org.candlepin.ApiClient client;
+
+    private String basePath;
+    private String endpoint;
+    private String method;
+    private String body;
+
+    private Map<String, String> pathParams;
+    private Map<String, String> headers;
+    private Map<String, List<String>> queryParams;
+
+
+    /**
+     * Builds a new Request instance which will use the specified ApiClient to perform the actual
+     * request upon execution.
+     *
+     * @param client
+     *  the org.candlepin.ApiClient instance to use to execute the request
+     *
+     * @throws IllegalArgumentException
+     *  if the provided client is null
+     */
+    public Request(org.candlepin.ApiClient client) {
```

Comment:
We can use just `ApiClient client` instead of using the full package

Reply:
Different ApiClients (`org.candlepin.ApiClient` and `org.candlepin.spec.bootstrap.client.ApiClient`). Due to the naming conventions used, one of the two must be fully qualified, and I opted to FQ the shorter of the two.

Reply:
Oh, that's my bad! I didn't realize they were different ApiClients.